### ISSUE TITLE
deprecate Http().createClientHttpsContext and provide alternative

### ISSUE THREAD

**Issue by robstoll** [akka/akka-http#2845]
AkkaSSLConfig was deprecated with 2.6.0, the migration guide points to https://doc.akka.io/docs/akka/current/stream/stream-io.html#tls but without showing how it has to be used in conjunction with akka-http. 

I am currently using `Http().createClientHttpsContext(...)` and look for an alternative (the new way to go) but cannot come up with something senseful as also `HttpsConnectionContext` is still using `val sslConfig: Option[AkkaSSLConfig]` meaning I would still stick with AkkaSSLConfig.

**Comment by raboof** [comment]
For the scaladsl, `HttpsConnectionContext` might be an alternative, passing `None` for `AkkaSSLConfig`.

For the javadsl I agree we need something nicer.

As for deprecating the methods that take an AkkaSSLConfig, we added this in https://github.com/akka/akka/issues/20214 to be able to disable SNI on a per-connection basis (instead of globally for the whole JVM). Let's double-check we indeed have a new way to do this first.
